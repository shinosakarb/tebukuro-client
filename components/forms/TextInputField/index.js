// @flow
import React from 'react'

type Props = {
  id: string,
  value: string | number,
  onChange: Function,
  onBlur: Function,
  errorMessages: ?[],
}

export default (props: Props) => (
  <div>
    <label htmlFor={props.id}>
      { props.id }
      <input type="text" {...props} />
      <span style={{ color: 'red', fontSize: '13px', paddingLeft: '5px' }} >
        { props.errorMessages }
      </span>
    </label>
  </div>
)
