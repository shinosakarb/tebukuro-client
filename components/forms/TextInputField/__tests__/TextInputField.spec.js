import React from 'react'
import { shallow } from 'enzyme'
import shallowToJson from 'enzyme-to-json'
import TextInputField from '../index'

const testProps = {
  id: 'name',
  value: 'event1',
  onChange: jest.fn(),
  onBlur: jest.fn(),
  errorMessages: ['nameは必須です。'],
}

describe('TextInput', () => {
  it('render event form component', () => {
    const wrapper = shallow(<TextInputField {...testProps} />)
    const tree = shallowToJson(wrapper)

    expect(tree).toMatchSnapshot()
  })
})
