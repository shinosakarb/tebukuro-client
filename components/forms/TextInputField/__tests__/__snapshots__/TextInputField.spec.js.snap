// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput render event form component 1`] = `
<div>
  <label
    htmlFor="name"
  >
    name
    <input
      errorMessages={
        Array [
          "nameは必須です。",
        ]
      }
      id="name"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
      value="event1"
    />
    <span
      style={
        Object {
          "color": "red",
          "fontSize": "13px",
          "paddingLeft": "5px",
        }
      }
    >
      nameは必須です。
    </span>
  </label>
</div>
`;
